server/models/Transaction.js

```diff
@@ -321,6 +322,34 @@ export default (Sequelize, DataTypes) => {
     }
   };
 
+  Transaction.prototype.setCurrency = async function (currency) {
+    // Nothing to do
+    if (currency === this.currency) {
+      return this;
+    }
+
+    const hostCurrencyFxRate = await getFxRate(currency, this.hostCurrency, this.createdAt);
+    // TODO: if Stripe transaction, we should better check if we have the rate stored there
+
+    // REMINDER: currency * hostCurrencyFxRate = amountInHostCurrency
+    // so: currency = amountInHostCurrency / hostCurrencyFxRate
+    const amount = Math.round(this.amountInHostCurrency / hostCurrencyFxRate);
+    const netAmountInCollectiveCurrency = Math.round(
+      (this.amountInHostCurrency +
+        this.platformFeeInHostCurrency +
+        this.hostFeeInHostCurrency +
+        this.paymentProcessorFeeInHostCurrency) /
+        hostCurrencyFxRate,
```

Comment:
@Betree how should we handle taxAmount there?

In which currency is it stored? And should it be removed from netAmountInCollectiveCurrency?

Reply:
`taxAmount` is always expressed with the same currency than `amount` (`transaction.currency`). We probably want to remove it from `netAmountInCollectiveCurrency`, I just checked and that's what we do in https://github.com/opencollective/opencollective-api/blob/master/server/models/Transaction.js#L624

Reply:
Are you sure it's not in hostCurrency? Because, the code at https://github.com/opencollective/opencollective-api/blob/master/server/models/Transaction.js#L624 assume it is.

Reply:
Affirmative. The way this amount is calculated in the contribution flow then in `createOrder` is `order.amount * taxPercent`. The code in https://github.com/opencollective/opencollective-api/blob/master/server/models/Transaction.js#L624 is therefor wrong to assume it's expressed in host currency, I missed that when I reviewed it.

I just checked in the DB and the ledger should be fine, we didn't face the case of a collective having a different currency than the host with VAT yet. That however surfaced 4 platform tips transactions that show platform tips and tax amounts don't interact well with each others. I'll create an issue for this, mentioning that we should update https://github.com/opencollective/opencollective-api/blob/master/server/models/Transaction.js#L624 too.

```sql
SELECT *
FROM "Transactions" t 
WHERE "taxAmount" < 0
AND "deletedAt" IS NULL
AND currency != 'EUR'
```

Reply:
@Betree thank you, I will then take this into account in my refactoring.